message_app_server/message_app/send_message.py

```diff
@@ -137,64 +154,67 @@ def is_typing(data):
+        attachment_name = str(created_at) + str(sender_id) + image_name # This is the image's id
         image_path = os.path.join(IMAGE_STORAGE_PATH, "regular_" + attachment_name)
         # Store the regular image to the system
         with open(image_path, "wb") as f:
             f.write(image)
 
         # Resize the image
-        resized_image = Image.open(image_path)
-        resized_image.thumbnail(THUMBNAIL_MAX_SIZE)
+        image_thumbnail = Image.open(image_path)
+        image_thumbnail.thumbnail(THUMBNAIL_MAX_SIZE)
 
         # Store resized version to the system
-        resized_path = os.path.join(IMAGE_STORAGE_PATH, "thumbnail_" + attachment_name)
-        resized_image.save(resized_path)
+        thumbnail_path = os.path.join(IMAGE_STORAGE_PATH, "thumbnail_" + attachment_name)
+        image_thumbnail.save(thumbnail_path)
 
         image_data = {
-            "regular_source": os.path.join("static", "user_images", "regular_" + attachment_name).replace("\\", "&#47;").replace(" ", "&#32;"), # Replace escape character with their codes
-            "thumbnail_source": os.path.join("static", "user_images", "thumbnail_" + attachment_name).replace("\\", "&#47;").replace(" ", "&#32;"), # Replace escape character with their codes
-            "width": resized_image.width,
-            "height": resized_image.height,
+            "regular_name": "regular_" + attachment_name,
+            "thumbnail_name": "thumbnail_" + attachment_name,
+            "has_attachment": True
         }
 
     # Update the database
-    user_id = User.select(username).id
     conversation = Conversations.select(conversation_id)
-    new_message = Messages(sender_id=user_id, content=message, created_at=created_at, conversation_id=conversation_id, attachment_name=attachment_name)
+    new_message = Messages(sender_id=sender_id, content=message, created_at=created_at, conversation_id=conversation_id, attachment_name=attachment_name)
     Messages.insert(new_message)
     conversation.last_message_id = new_message.id # Manually set the last_message_id for now
     DB.session.commit()
 
+    # Get sender_name
+    sender_name = User.select(user_id=sender_id).username
+
     # Return data
-    return_data = {
+    payload = {
         "id": new_message.id,
-        "conversation_id": conversation_id,
-        "username": username,
-        "message": message,
-        "created_at": created_at
+        "sender_id": sender_id,
+        "sender_name": sender_name,
+        "content": message,
+        "created_at": created_at,
+        "has_attachment": False
     }
 
     if image:
-        return_data.update(image_data)
+        payload.update(image_data)
 
-    emit("message_handler_client", return_data, room=conversation_id)
+    emit("updateLastMessageID", {"conversation_id": conversation_id, "last_message_id": new_message.id}, room=conversation_id)
```

Comment:
So before this PR, if we sent an image, it was not considered as the last message?